packages/infra/lib/api-stack/api-service.ts

```diff
@@ -108,6 +108,9 @@ export function createAPIService(
           AWS_REGION: props.config.region,
           TOKEN_TABLE_NAME: dynamoDBTokenTable.tableName,
           API_URL: `https://${props.config.subdomain}.${props.config.domain}`,
+          ...(props.config.apiGatewayUsagePlanId
+            ? { API_GW_USAGE_PLAN_ID: props.config.apiGatewayUsagePlanId }
+            : {}),
```

Comment:
nit: why not follow the existing pattern below? ie

```
 ...(props.config.apiGatewayUsagePlanId && {
            API_GW_USAGE_PLAN_ID: props.config.apiGatewayUsagePlanId,
          }),
```

Reply:
Yeah, as I was writing it, it felt wrong. Everywhere else I remember I use this new one, not sure why we have this pattern on infra.

It works if the prop being checked is `undefined`, but everything else results in sending the falsy value to be decomposed (false, null, empty string).

By building a full ternary it looks more clear what the the values can be, and we don't promote a pattern than can fail depending on the value being used.

Reply:
https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4